### PR TITLE
Change margin on day numbers so that they are centered even with dots

### PR DESCRIPTION
**Pull request by lainsce**
If a day has no events, it shouldn't reserve space for dots, as such, this PR solves this problem which also as a bonus helps the selection circle have days in the middle correctly.

Screenshot:
Screenshot from 2019-09-21 00 38 09

### CODE REVIEW

File: src/Widgets/calendar/GridDay.vala [hunk 1]
```diff
@@ -115,6 +115,8 @@ public class DateTime.Widgets.GridDay : Gtk.EventBox {
                         event_dots[event_uid] = event_dot;
 
                         event_grid.add (event_dot);
+                        grid.attach (event_grid, 0, 1);
```
Comment (tintou) [inline src/Widgets/calendar/GridDay.vala]: You're calling attach several times here, you can check `if (event_grid.get_parent () != null)` before calling it

File: src/Widgets/calendar/GridDay.vala [hunk 2]
```diff
@@ -115,6 +115,8 @@ public class DateTime.Widgets.GridDay : Gtk.EventBox {
...
+                        grid.margin_top = 6;
```
Comment (tintou) [inline src/Widgets/calendar/GridDay.vala]: Put this at grid creation time, not here

File: src/Widgets/calendar/GridDay.vala (hunk 1 above)
Comment (lainsce) [reply src/Widgets/calendar/GridDay.vala]: Done. See the newest commit. :smile:

File: src/Widgets/calendar/GridDay.vala (hunk 2 above)
Comment (lainsce) [reply src/Widgets/calendar/GridDay.vala]: Done.

**Comment by lainsce** [comment]
Review points addressed, any more stuff to do before merge?

**Comment by danrabbit** [comment]
Week numbers are no longer aligned in this branch

Screenshot from 2019-09-23 10 14 48@2x

**Comment by lainsce** [comment]
Oops, that'll be fixed as well as soon as I can. I always forget about that part of the UI. :laughing:

**Comment by lainsce** [comment]
Week numbers aligned.

Screenshot from 2019-09-23 15 31 41

File: src/Widgets/calendar/GridDay.vala [hunk 3]
```diff
@@ -56,21 +57,30 @@ public class DateTime.Widgets.GridDay : Gtk.EventBox {
 
         label = new Gtk.Label (null);
 
+        event_dots = new Gee.HashMap<string, Gtk.Widget> ();
```
Comment (danrabbit) [inline src/Widgets/calendar/GridDay.vala]: Why did you move these 4 lines here? Is it necessary?

File: src/Widgets/calendar/GridDay.vala [hunk 4]
```diff
@@ -56,21 +57,30 @@ public class DateTime.Widgets.GridDay : Gtk.EventBox {
...
+
+        model.events_added.connect (add_event_dots);
+        model.events_removed.connect (remove_event_dots);
+
         event_grid = new Gtk.Grid ();
         event_grid.halign = Gtk.Align.CENTER;
         event_grid.height_request = 6;
 
-        var grid = new Gtk.Grid ();
```
Comment (danrabbit) [inline src/Widgets/calendar/GridDay.vala]: It doesn't look like you're accessing this grid from outside of the `construct` block. Why did you make it available to the whole class?

File: src/Widgets/calendar/GridDay.vala [hunk 5]
```diff
@@ -56,21 +57,30 @@ public class DateTime.Widgets.GridDay : Gtk.EventBox {
...
+        grid = new Gtk.Grid ();
         grid.halign = grid.valign = Gtk.Align.CENTER;
         grid.attach (label, 0, 0);
-        grid.attach (event_grid, 0, 1);
+
+        if (event_grid.get_parent () == null) {
```
Comment (danrabbit) [inline src/Widgets/calendar/GridDay.vala]: Do we have reason to believe that event_grid already has a parent at construction time? It doesn't look like there's any other place where event_grid is attached or added to another widget

Comment (lainsce) [reply src/Widgets/calendar/GridDay.vala]: It seems so. If I set it to `event_grid.get_parent () != null` then no dots are shown for any events.

File: src/Widgets/calendar/GridDay.vala (hunk 4 above)
Comment (lainsce) [reply src/Widgets/calendar/GridDay.vala]: So I could use it on the add_event_dots method.

File: src/Widgets/calendar/GridDay.vala (hunk 3 above)
Comment (lainsce) [reply src/Widgets/calendar/GridDay.vala]: Needed to capture the events into the hashmap before doing the grid stuff for the days.

**Comment by lainsce** [comment]
Hopefully my explanations make sense :thinking:

File: src/Widgets/calendar/GridDay.vala (hunk 5 above)
Comment (danrabbit) [reply src/Widgets/calendar/GridDay.vala]: Right, because when this class is constructed `event_grid` should not yet have a parent and there's nowhere else in this class where `event_grid` is attached or added to the grid. So this code might as well say `if (1 == 1)`

Comment (lainsce) [reply src/Widgets/calendar/GridDay.vala]: I see. I'll fix it right away to not have the `if` block.

**Comment by lainsce** [comment]
@danrabbit Anything more that is needed here?

**Comment by lainsce** [comment]
@tintou Have made fixes for your review points as well before Dan's review point fixes.

**Comment by lainsce** [comment]
> I'm not sure this is an improvement since it just makes days with events look off center instead of days with no events looking off center.

I'm not sure why that's happening over there, maybe HiDPI; see this screenshot where I use the PR, which has the focus circle and day text here in perfect alignment as it should be:
Screenshot from 2019-10-22 14 08 39

**Comment by lainsce** [comment]
I also changed PR name to reflect current PR codebase

**Comment by danrabbit** [comment]
Right so now the numbers are centered, but the whole area (the numbers and the dots) are closer to the bottom than the top.

So really this PR doesn't smartly do anything it all. It moves margin from one side to the other.

I don't think it would require anything hacky at all to draw the background around the number only. It looks like we'd need to remove some CSS we currently add to change the dots to white and then add something like:

```
.circular {
    background: transparent;
}

.circular:selected label {
    background: @selected_bg_color;
    border-radius: 50%;
    min-height: 1.5em;
    min-width: 1.5em;
    padding: 1px;
}
```

**Comment by lainsce** [comment]
I'll use that css for the next commit

**Comment by Dirli** [comment]
https://github.com/Dirli/wingpanel-indicator-datetime/commit/cb3c803eafcf7198bbcb23318dbeaae670e5fd27
Maybe this?

**Comment by danrabbit** [comment]
Closing in favor of #222
